File: src/structure/bundle/createBundleInfoItem.ts
import path from "node:path";
import {StructureItem} from "../structure.js";
import {Configuration} from "../../configuration/configuration.js";
import {createSourceFileInfoItem} from "../sourceFile/createSourceFileInfoItem.js";

export function createBundleInfoItem(
    configuration: Configuration,
): StructureItem {
    const {outputFileName} = configuration

    const fileName = outputFileName
        ? path.basename(outputFileName)
        : "bundle.kt"

    const syntheticItem = createSourceFileInfoItem("", configuration)

    return {
        ...syntheticItem,
        fileName,
    }
}

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/structure/bundle/createBundleInfoItem.kt
package io.github.sgrishchenko.karakum.structure.bundle

import io.github.sgrishchenko.karakum.configuration.Configuration
import io.github.sgrishchenko.karakum.structure.StructureItem
import io.github.sgrishchenko.karakum.structure.sourceFile.createSourceFileInfoItem
import node.path.path

fun createBundleInfoItem(
    configuration: Configuration,
): StructureItem {
    val outputFileName = configuration.outputFileName

    val fileName = if (outputFileName != null) path.basename(outputFileName) else "bundle.kt"

    val syntheticItem = createSourceFileInfoItem("", emptyArray(), configuration)

    return StructureItem(
        fileName = fileName,
        `package` = syntheticItem.`package`,
        moduleName = syntheticItem.moduleName,
        qualifier = syntheticItem.qualifier,
        hasRuntime = syntheticItem.hasRuntime,
        imports = syntheticItem.imports,
    )
}
